### PR TITLE
Config: Extract languages to shared config and remove obsolete muse_supported_themes

### PR DESCRIPTION
**Pull request by gziolo**
We repeat across all config files so many values. We could easily mitigate that issue by using already present `_shared.json` config file introduced by @mjangda in #876. This PR tries to slim down config files by extracting `muse_supported_themes` and `languages` fields. We probably can open the follow up PR with more changes.

### Testing
1. Run `npm test` and make sure everything passes.
2. Run `http://calypso.localhost:3000/me/account` and make sure you still can change language for your interface :)
3. Go to your site's setting page and check if you can change language of your site.

### CODE REVIEW

File: config/_shared.json [hunk 1]
```diff
@@ -1,3 +1,170 @@
...
+		{ "value": 454, "langSlug": "bs", "name": "bs - Bosanski" },
+		{ "value": 10, "langSlug": "ca", "name": "ca - Català" },
+		{ "value": 425, "langSlug": "ce", "name": "ce - Нохчийн" },
+		{ "value": 11, "langSlug": "cs", "name": "cs - Česky" },
+		{ "value": 12, "langSlug": "csb", "name": "csb - Kaszëbsczi" },
+		{ "value": 426, "langSlug": "cv", "name": "cv - Чӑваш" },
+		{ "value": 13, "langSlug": "cy", "name": "cy - Cymraeg" },
+		{ "value": 14, "langSlug": "da", "name": "da - Dansk" },
+		{ "value": 15, "langSlug": "de", "name": "de - Deutsch", "popular": 4 },
+		{ "value": 427, "langSlug": "dv", "name": "dv - ދިވެހިބަސް" },
```
Comment (gziolo) [inline config/_shared.json]: Can someone double check if that works properly for users? Everything else displays properly for me, but this one seems to be broken.

**Comment by mtias** [comment]
I haven't tested the language ones, but it's looking good to me.

**Comment by gziolo** [comment]
It turned out `muse_supported_themes` value is no longer referenced in code. I removed it. @sirbrillig can you confirm that change?

@akirk can you double check language related change?

**Comment by sirbrillig** [comment]
@gziolo wow, I did some digging and the use-case for `muse_supported_themes` was removed on August 4, 2015. (It had been added on Mar 4, 2015 to prevent showing any customize link in the sidebar (client/my-sites/sidebar/sidebar.jsx) when Muse would not be available, because the native customizer experience at mobile width was so poor.)

I'd say removing it is fine.

**Comment by gziolo** [comment]
@mtias can we land this one? It looks like there are no blockers.

**Comment by mtias** [comment]
Yeah, let's 🚢

**Pull request merged**